model_compression_toolkit/core/keras/keras_implementation.py

```diff
@@ -137,11 +141,16 @@ def model_builder(self,
         Returns:
             A tuple of the Keras model that was built and an UserInformation object.
         """
-        return model_builder(graph,
-                             mode,
-                             append2output,
-                             fw_info,
-                             return_float_outputs=return_float_outputs)
+
+        keras_builders = {ModelBuilderMode.QUANTIZED: QuantizedKerasModelBuilder,
+                          ModelBuilderMode.FLOAT: FloatKerasModelBuilder,
+                          ModelBuilderMode.MIXEDPRECISION: MixedPrecisionKerasModelBuilder}
+        builder = keras_builders.get(mode)
+        return builder(graph=graph,
```

Comment:
Can't we remove the enum now? Maybe we should write a different function in fw_implemtation for each builder?

Reply:
I thought about it, but it seemed clearer to me this way. I'm down with both ways